packages/js/product-editor/src/components/header/header.tsx

```diff
@@ -1,14 +1,76 @@
 /**
  * External dependencies
  */
-import { __ } from '@wordpress/i18n';
+import { Product } from '@woocommerce/data';
+import { Button } from '@wordpress/components';
+import { useDispatch, useSelect } from '@wordpress/data';
 import { createElement } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+import { navigateTo, getNewPath } from '@woocommerce/navigation';
+
+/**
+ * Internal dependencies
+ */
+import { AUTO_DRAFT_NAME } from '../../utils';
 
 export type HeaderProps = {
-	title: string;
+	productId: number;
+	productName: string;
 };
 
-export function Header( { title }: HeaderProps ) {
+export function Header( { productId, productName }: HeaderProps ) {
+	const { isProductLocked, isSaving, product } = useSelect(
+		( select ) => {
+			const { isSavingEntityRecord, getEditedEntityRecord } =
+				select( 'core' );
+			const { isPostSavingLocked } = select( 'core/editor' );
+			return {
+				isProductLocked: isPostSavingLocked(),
+				isSaving: isSavingEntityRecord(
+					'postType',
+					'product',
+					productId
+				),
+				product: getEditedEntityRecord(
```

Comment:
Did you actually want to return `productName` only here, instead of the entire `product`.
Sorry for harping on this, but return the product would trigger a re-render on every product update, although we only care about the `name`.

Reply:
Should I use a different function for that, because I could not find it within the EntityRecord selectors.
Or do you meant calling the `const { name } = getEditedEntityRecord(...)` like that and just return the name of the product like `{ productName: name }`?

Reply:
Yeah that second option.

Reply:
[1d0165e](https://github.com/woocommerce/woocommerce/pull/37152/commits/1d0165e875634df4d66ba20d715574f9e5423d29)